core/services/standardcapabilities/delegate.go

```diff
@@ -203,7 +205,27 @@ func (d *Delegate) ServicesForSpec(ctx context.Context, spec job.Job) ([]job.Ser
 	}
 
 	if spec.StandardCapabilitiesSpec.Command == commandOverrideForCustomComputeAction {
-		computeSrvc := compute.NewAction(log, d.registry)
+		if d.gatewayConnectorWrapper == nil {
+			return nil, errors.New("gateway connector is required for custom compute capability")
+		}
+		connector := d.gatewayConnectorWrapper.GetGatewayConnector()
```

Comment:
Nit: this is declared here but used for the first time on line 223 -- why not declare it as close as possible to its first use? This increases cohesion :)